source/language/pulses.rst

```diff
@@ -92,3 +79,217 @@ circuit. Most of the time symbols in the ``defcal`` table will also have
 corresponding ``gate`` definitions. However, if a user provides a ``defcal`` for a symbol
 without a corresponding ``gate``, then we treat such operations like the ``opaque`` gates
 of prior versions of OpenQASM.
+
+OpenPulse instructions
+======================
+
+In addition to OpenQASM instructions, ``defcal`` blocks may contain OpenPulse 
+instructions using a text-based version of OpenPulse defined here. OpenPulse is 
+a JSON format described in the paper 
+`Qiskit Backend Specifications for OpenQASM and OpenPulse Experiment <https://arxiv.org/abs/1809.03452>`_.
+
+The text format described here has several advantages over the equivalent JSON 
+format:
+
+- It is more readable
+- Absolute time is handled through built-in timekeeping instructions
+- Gates and classical instructions can be mixed in with the pulses to create far richer calibrations
+
+Channels
+--------
+
+Channels (sometimes referred to as "frames") describe the different forms of
+classically-controlled stimulus fields used to interact with a qubit. There are
+two kinds of channels: input channels, which send signal into the system to
+drive the qubit, and output channels, which measure signal from the system to
+measure qubits.
+
+Each channel has a user-specified local oscillator frequency which is assumed to
+have been specified outside of the context of OpenQASM.
```

Comment:
Do we also expect to specify other properties like the sample rate for the `channel`? Especially useful if we do delayed resolve to samples [as per this discussion](https://github.com/Qiskit/openqasm/pull/109/files#r550954765).